packages/table/src/useLeafyGreenTable/useLeafyGreenTable.tsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import { useVirtual } from 'react-virtual';
+import { Table, useReactTable } from '@tanstack/react-table';
+import { Row } from '@tanstack/react-table';
+import PropTypes from 'prop-types';
+
+import Checkbox from '@leafygreen-ui/checkbox';
+
+import { LeafyGreenTableOptions, LGRowData } from './useLeafyGreenTable.types';
+import {
+  LeafyGreenTable,
+  LGColumnDef,
+  LGTableDataType,
+  VirtualizerValues,
+} from '.';
+
+const selectColumnConfig: LGColumnDef<LGRowData> = {
+  id: 'select',
+  size: 14, // match checkbox width
+  header:
+    // eslint-disable-next-line react/display-name
+    ({ table }: { table: Table<LGTableDataType<LGRowData>> }) => (
+      <Checkbox
+        checked={table.getIsAllRowsSelected()}
+        indeterminate={table.getIsSomeRowsSelected()}
+        onChange={table.getToggleAllRowsSelectedHandler()}
+        aria-label="Select all rows"
+      />
+    ),
+  cell:
+    // eslint-disable-next-line react/display-name
+    ({
+      row,
+      table,
+    }: {
+      table: Table<LGTableDataType<LGRowData>>;
+      row: Row<LGTableDataType<LGRowData>>;
+    }) => (
+      <Checkbox
+        checked={row.getIsSelected()}
+        indeterminate={row.getIsSomeSelected()}
+        onChange={row.getToggleSelectedHandler()}
+        aria-label={`Select row ${row.index}`}
+        // Don't animate if _all_ rows have been checked (usually, if header row is clicked). Not the _best_ check, but it mostly works
```

Comment:
What's this comment?

Reply:
@brookescarlett When a user clicks the "select all" button, we don't want all the other checkboxes to animate. So we disable animation when _all_ the rows are selected

Reply:
ah ok makes sense